great_expectations/dataset/file_dataset.py

```diff
@@ -0,0 +1,364 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Oct 20 13:49:29 2018
+
+@author: anhol
+"""
+
+from __future__ import division
+
+from .base import DataFile, Dataset
+import re
+import numpy as np
+from itertools import compress
+from six import PY3
+from .util import DocInherit, parse_result_format
+import inspect
+from functools import wraps
+import hashlib
+import os
+import csv
+import json
+import jsonschema
+
+
+class MetaFileDataset(DataFile):
+    """MetaFileDataset is a thin layer between Dataset and FileDataset.
+    This two-layer inheritance is required to make @classmethod decorators work.
+    Practically speaking, that means that MetaFileDataset implements \
+    expectation decorators, like `column_map_expectation` and `column_aggregate_expectation`, \
+    and FileDataset implements the expectation methods themselves.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(MetaFileDataset, self).__init__(*args, **kwargs)
+        
+        
+    @classmethod
+    
+    def file_map_expectation(cls, func):
```

Comment:
I think file-map is confusing (this is done on a per-line basis, right, so we're mapping over lines in a (non-binary) file, vs over files)

Reply:
Yes, I agree it is confusing. I will plan to change this naming convention to something like file_lines_map.